### PR TITLE
retain existing AKS SNAT implementation

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: A follow-up to 
https://github.com/Azure/acs-engine/pull/3916

Allows acs-engine to bake ip-masq-agent SNAT implementation before merging into AKS.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
retain existing AKS SNAT implementation
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@JunSun17 @CecileRobertMichon FYI this PR would enable acs-engine to bake the ip-masq-agent changes longer before bringing them into AKS

File: pkg/acsengine/params_k8s.go [hunk 1]
```diff
@@ -259,14 +259,18 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				CloudProviderRateLimitBucket: kubernetesConfig.CloudProviderRateLimitBucket,
 			})
 			addValue(parametersMap, "kubeClusterCidr", kubernetesConfig.ClusterSubnet)
-			if properties.OrchestratorProfile.IsAzureCNI() {
```
Comment (jackfrancis) [inline pkg/acsengine/params_k8s.go]: Pro tip: side diff view makes this even possible to grok

**Comment by jackfrancis** [comment]
The key line in the acs-engine E2E tests are:

```
Azure Container Cluster using the Kubernetes Orchestrator regardless of agent pool type 
  should have ip-masq-agent running
  /go/src/github.com/Azure/acs-engine/test/e2e/kubernetes/kubernetes_test.go:430

$ kubectl get pods -n kube-system -o json

$ kubectl get pods -n kube-system -o json

$ kubectl get pods -n kube-system -o json

$ kubectl get pods -n kube-system -o json
STEP: Ensuring that the correct resources have been applied

$ kubectl get pods -n kube-system -o json
•
```

i.e., `ip-masq-agent` addon is the responsible vector for enforcing SNAT on the vms for acs-engine-built clusters

File: pkg/acsengine/params_k8s.go [hunk 2]
```diff
@@ -259,14 +259,18 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
...
+					} else {
+						addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
+					}
 				} else {
-					addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
+					addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
 				}
 			} else {
-				addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
+				addValue(parametersMap, "kubernetesNonMasqueradeCidr", kubernetesConfig.KubeletConfig["--non-masquerade-cidr"])
```
Comment (JunSun17) [inline pkg/acsengine/params_k8s.go]: i think we do not need to set this parameter for AKS, this parameters seems are only used in ace engine implementation.

Comment (CecileRobertMichon) [inline pkg/acsengine/params_k8s.go]: Where is `kubernetesNonMasqueradeCidr` used for hosted masters? Looking at the diff for https://github.com/Azure/acs-engine/pull/3916 it seems to only be used for masters

**Comment by jackfrancis** [comment]
@CecileRobertMichon @JunSun17 you were both right, thanks!

**Pull request merged**